### PR TITLE
feat: support for subscribing to channels in search results

### LINKED ISSUE #509: Quick Subscribe: Showing 'Subscribe' button on search results page
Hello,

It'd be great if I could see the Subscribe button displayed on the search results page in the same or similar way Youtube does.

image

Thank you!

### PR DESCRIPTION
**Pull request by Bnyro**
closes #509

### CODE REVIEW

File: src/components/ChannelItem.vue [hunk 1]
```diff
@@ -1,37 +1,72 @@
...
+            this.subscribed = await this.fetchSubscriptionStatus(this.channelId);
+            console.log(this.subscribed);
+        },
+        subscribeHandler() {
+            this.toggleSubscriptionState(this.channelId, this.subscribed).then(success => {
+                if (success) this.subscribed = !this.subscribed;
+            });
+        },
     },
-});
```
Comment (FireMasterK) [inline src/components/ChannelItem.vue:36]: We should not change this to the legacy Options API, we should be using the [Composition API](https://vuejs.org/guide/extras/composition-api-faq.html) like before ideally.

Ideally, we should migrate our existing code to the composition api.

**Review by FireMasterK — APPROVED** [review]

File: src/components/ChannelItem.vue (hunk 1 above)
Comment (Bnyro) [reply src/components/ChannelItem.vue:36]: Looks awesome, quite similar to Svelte and a lot less boilerplate code :)

Comment (Bnyro) [reply src/components/ChannelItem.vue]: How should we be using `this` now? The Vue docs say that we should migrate to replacing mixins with [Composable functions](https://vuejs.org/guide/reusability/composables), but that's certainly out of scope for this PR.

File: src/components/ChannelItem.vue [hunk 2]
```diff
@@ -23,15 +23,42 @@
...
+const channelId = props.item.url.substr(-24);
+
+onMounted(() => {
+    updateSubscribedStatus();
+});
+async function updateSubscribedStatus() {
+    subscribed.value = await this.fetchSubscriptionStatus(channelId);
+}
+function subscribeHandler() {
+    this.toggleSubscriptionState(channelId, subscribed.value).then(success => {
```
Comment (Bnyro) [inline src/components/ChannelItem.vue:60]: `this` not found

File: src/components/ChannelItem.vue [hunk 3]
```diff
@@ -23,15 +23,42 @@
...
     },
 });
+const subscribed = ref(null);
+const channelId = props.item.url.substr(-24);
+
+onMounted(() => {
+    updateSubscribedStatus();
+});
+async function updateSubscribedStatus() {
+    subscribed.value = await this.fetchSubscriptionStatus(channelId);
```
Comment (Bnyro) [inline src/components/ChannelItem.vue:57]: `this` not found

File: src/components/ChannelItem.vue (hunk 1 above)
Comment (Bnyro) [reply src/components/ChannelItem.vue:36]: Continuing the use of the composition API would require major refactors, which I don't intend to do with this PR.

**Pull request merged**